editquality/codegen/generate.py

```diff
@@ -1,19 +1,24 @@
 import jinja2
 
-from . import config
 
+def generate(variables, templates_path, main_template):
+    """
+    :Parameters:
+        variables : dict
+            Template parameters, passed through.
+        templates_path : str
+            Root directory for transclusions.
+        main_template : str
+            Contents of the main template.
 
-def generate(config_path, templates_path):
-    variables = config.load_config(config_path)
-
-    # TODO: map config into domain-specific preprocessed values
+    Returns the rendered output.
+    """
 
     env = jinja2.Environment(
         loader=jinja2.FileSystemLoader(templates_path),
         lstrip_blocks=True,
         trim_blocks=True
     )
```

Comment:
Could use BaseLoader

Reply:
I'm planning to introduce partials, which require a filesystem lookup...

Reply:
What do you mean by partials?  Is this like including one j2 template in another?  If so, I think I see where you are coming from.

Reply:
Exactly.  Here's the documentation, http://jinja.pocoo.org/docs/2.10/templates/#include

I'm still fine with making the testability change and passing string template contents, cos I like that the calling function has more control.  Maybe we should allow overriding the loader, in the future.

Reply:
OK this makes sense to me.